src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SystemProxyInfo.Windows.cs

```diff
@@ -14,7 +14,7 @@ public static IWebProxy ConstructSystemProxy()
                 HttpSystemProxy.TryCreate(out proxy);
             }
 
-            return proxy;
+            return proxy ?? new HttpNoProxy();
```

Comment:
Should we instead check for null at the call site and create this HttpNoProxy there instead?

Reply:
By "call site" do you mean SocketsHttpHandler which calls SystemProxyInfo.ConstructSystemProxy() ?

I don't want to do that.  I want ConstructSystemProxy() to always return a non-null IWebProxy object representing the current platform/system proxy settings.

And per my comment, https://github.com/dotnet/corefx/pull/37306#issuecomment-488171770, i will doing another PR to change this to a Singleton pattern (probably `Lazy<T>` for the return of the ConstructSystemProxy() method as part of finishing #36553

Reply:
> By "call site" do you mean SocketsHttpHandler which calls SystemProxyInfo.ConstructSystemProxy() ?

Yes.

> I don't want to do that. I want ConstructSystemProxy() to always return a non-null IWebProxy object representing the current platform/system proxy settings.

Ok.